### DISCUSSION TITLE
Batched Faster Whisper, how to use "vad_onset" and "vad_offset"

### DISCUSSION THREAD

**Discussion by toanhn** [SYSTRAN/faster-whisper#1047] [Q&A]
I use Batched Faster Whisper to transcribe audio files.
If I use these default values vad_onset: float = 0.500, vad_offset: float = 0.363 then some audio files are missed sentences, I change to vad_onset=0.03, vad_offset=0.03 then it works well, but it cannot work well for other audio files.

How can I use vad_onset and vad_offset for all audio files?

**Comment by Hikari-Tsai** [comment]
Those names belong to an older VAD interface. In current faster-whisper the corresponding Silero settings are `threshold` (start/positive speech threshold) and `neg_threshold` (end/silence threshold):

```python
segments, info = model.transcribe(
    audio,
    vad_filter=True,
    vad_parameters={
        "threshold": 0.5,
        "neg_threshold": 0.35,
        "min_silence_duration_ms": 500,
        "speech_pad_ms": 300,
    },
)
```

There is no threshold pair that is optimal for every recording. A value of `0.03` classifies almost any low-level sound as speech, which may recover quiet sentences but also defeats most of the protection against music, noise and hallucinations.

Normalize the input conditions first, then tune on a representative validation set. For quiet speech, lower `threshold` gradually (for example 0.5 → 0.4 → 0.3) and retain hysteresis by keeping `neg_threshold` lower than `threshold`. Use `speech_pad_ms` to avoid clipping word edges and `min_silence_duration_ms` to control splitting. If missing speech remains costly, disable VAD for those files rather than using an extremely low universal threshold.
